File: game.scala
import scala.io.StdIn.readLine
import scala.util.Random
import scala.util.control.Breaks._
object Irritext {
    def main(args: Array[String]): Unit = {
        println("Hello and welcome to 'What number is this!?'\n")

        var score = 0
        var a = Random.nextInt(10)
        var b = Random.nextInt(10)
        
        breakable {
        while (true) {

            println(s"If the number $a is the number $b, then what is the number ${Random.nextInt(10)}!? ")
            var guess = readLine()

            if (guess == a.toString()) {

                println(s"Correct... Your score is $score points.")
                break
            }

            else {

                println("Wrong...")
                score -= Random.nextInt(10)
            }
            b = a
            a = Random.nextInt(10)
            }
        }
    }
}

File: game.py
# What number is this!?
import random
print("Hello and welcome to 'What number is this!?'\n")

score = 0
a = random.randint(1, 10)
b = random.randint(1, 10)

while True:

    print(f"If the number {a} is the number {b}, then what is the number {random.randint(1, 10)}!? ")
    guess = input()

    if guess == str(a):

        print(f"Correct... Your score is {score} points.")
        break

    else:

        print("Wrong...")
        score -= random.randint(5, 10)

    b = a
    a = random.randint(1, 10)
